beets/dbcore/query.py

```diff
@@ -93,7 +110,21 @@ def __eq__(self, other) -> bool:
         return type(self) == type(other)
 
     def __hash__(self) -> int:
-        return 0
+        """Minimalistic default implementation of a hash.
+
+        Given the implementation if __eq__ above, this is
+        certainly correct.
+        """
+        return hash(type(self))
+
+
+class NamedQuery(Query):
+    """Non-field query, i.e. the query prefix is not a field but identifies the
+    query class.
+    """
+    @abstractmethod
+    def __init__(self, pattern):
+        ...
```

Comment:
This is interesting: it's a good way to capture what is in common between queries that have a pattern but no field (i.e., are "half" of a `FieldQuery`). This makes sense, and it's a good way to deal with the typing requirements.